addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -64,8 +66,34 @@ def _are_you_inside(self):
 
     @api.depends('user_ids.im_status')
     def _compute_available_operator_ids(self):
+        session_by_operator = {}
+        operator_in_call = self.env["res.partner"]
+        if self.concurent_session:
+            session_by_operator = dict(
+                self.env["discuss.channel"]._read_group(
+                    [
+                        ("livechat_channel_id", "!=", False),
+                        ("livechat_active", "=", True),
+                        ("last_interest_dt", ">", fields.Datetime.now() - timedelta(minutes=30)),
+                    ],
+                    groupby=["livechat_operator_id"],
+                    aggregates=["__count"],
+                )
+            )
+            operator_in_call = (
+                self.env["discuss.channel.rtc.session"]
+                .search([("partner_id", "in", self.user_ids.partner_id.ids)])
+                .partner_id
+            )
         for record in self:
-            record.available_operator_ids = record.user_ids.filtered(lambda user: user._is_user_available())
+            record.available_operator_ids = record.user_ids.filtered(
+                lambda user: user._is_user_available()
+                and (
+                    not session_by_operator.get(user.partner_id)
+                    or session_by_operator.get(user.partner_id) < self.concurent_session_number
```

Comment:
```suggestion
                    or session_by_operator[user.partner_id] < self.concurent_session_number
```
We know for a fact that partner is in the dict keys. Otherwise this code could lead to "< not supported between NoneType and datetime.datetime".